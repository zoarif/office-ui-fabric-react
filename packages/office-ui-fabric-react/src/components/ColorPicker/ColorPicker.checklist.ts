import { ChecklistStatus } from '../../demo/ComponentStatus/ComponentStatus.types';

export const ColorPickerStatus = {
  keyboardAccessibilitySupport: ChecklistStatus.unknown,
  markupSupport: ChecklistStatus.unknown,
  highContrastSupport: ChecklistStatus.fail,
  rtlSupport: ChecklistStatus.pass,
  testCoverage: ChecklistStatus.fair
};