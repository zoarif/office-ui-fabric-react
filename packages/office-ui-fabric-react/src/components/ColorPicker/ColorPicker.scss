@import '../../common/common';

.root {
  position: relative;
  max-width: 300px;
}

.panel {
  padding: 16px;
}

.colorRect {
  position: relative;
  margin-bottom: 10px;

  @include high-contrast {
    -ms-high-contrast-adjust: none;
  }
}

.rectContainer {
  position: relative;
}

.capture {
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
  background: rgba(255,0,0,.1);
}

.rectContainer.rectContainerIsAdjusting .capture {
  position: fixed;
}

.table {
  table-layout: fixed; // fix IE
  width: 100%;
}
.tableHexCell {
  width: 25%;
}

.thumb {
  position: absolute;
  width: 20px;
  height: 20px;
  background: white;
  border: 1px solid rgba(255,255,255,.8);
  border-radius: 50%;
  box-shadow: 0 0 15px -5px black;
  transform: translate(-50%, -50%);

  &.thumbIsSlider {
    top: 50%;
  }
}

.light {
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  background: linear-gradient(to right, white 0%, transparent 100%);
}

.dark {
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  background: linear-gradient(to bottom, transparent 0, #000 100%);
}

.slider {
  position: relative;
  height: 20px;
  margin-bottom: 5px;
  border: 1px solid $ms-color-neutralLight;
  box-sizing: border-box;

  &.colorSliderIsHue {
    background: linear-gradient(to left,red 0,#f09 10%,#cd00ff 20%,#3200ff 30%,#06f 40%,#00fffd 50%,#0f6 60%,#35ff00 70%,#cdff00 80%,#f90 90%,red 100%);
  }

  &.colorSliderIsAlpha {
    background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAYAAACNMs+9AAAAJUlEQVQYV2N89erVfwY0ICYmxoguxjgUFKI7GsTH5m4M3w1ChQC1/Ca8i2n1WgAAAABJRU5ErkJggg==);
  }
}

.sliderOverlay {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
}

.input {
  width: 100%;
  border: none;
  box-sizing: border-box;
  height: 30px;

  &:global(.ms-TextField) {
    padding-right: 2px;
  }

  :global(.ms-TextField-field) {
    min-width: auto;
    padding: 5px;
    text-overflow: clip;
  }
}

