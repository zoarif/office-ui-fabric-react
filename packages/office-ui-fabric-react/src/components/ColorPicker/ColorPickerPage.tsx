import * as React from 'react';
import {
  ExampleCard,
  IComponentDemoPageProps,
  ComponentPage,
  PropertiesTableSet,
  PageMarkdown,
} from '@uifabric/example-app-base';
import { ColorPickerBasicExample } from './examples/ColorPicker.Basic.Example';
import { ComponentStatus } from '../../demo/ComponentStatus/ComponentStatus';
import { ColorPickerStatus } from './ColorPicker.checklist';

const ColorPickerBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/ColorPicker/examples/ColorPicker.Basic.Example.tsx') as string;

export class ColorPickerPage extends React.Component<IComponentDemoPageProps, {}> {
  public render(): JSX.Element {
    return (
      <ComponentPage
        title='ColorPicker'
        componentName='ColorPickerExample'
        componentUrl='https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/ColorPicker'
        exampleCards={
          <ExampleCard
            title='Default ColorPicker'
            code={ ColorPickerBasicExampleCode }
          >
            <ColorPickerBasicExample />
          </ExampleCard>
        }
        propertiesTables={
          <PropertiesTableSet
            sources={ [
              require<string>('!raw-loader!office-ui-fabric-react/src/components/ColorPicker/ColorPicker.types.ts')
            ] }
          />
        }
        overview={
          <PageMarkdown>
            { require<string>('!raw-loader!office-ui-fabric-react/src/components/ColorPicker/docs/ColorPickerOverview.md') }
          </PageMarkdown>
        }
        isHeaderVisible={ this.props.isHeaderVisible }
        componentStatus={
          <ComponentStatus
            { ...ColorPickerStatus }
          />
        }
      />
    );
  }
}