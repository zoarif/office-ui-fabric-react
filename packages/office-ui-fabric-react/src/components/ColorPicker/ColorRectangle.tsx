import * as React from 'react';
import {
  BaseComponent,
  assign,
  css,
  createRef
} from '../../Utilities';
import {
  IColor,
  MAX_COLOR_SATURATION,
  MAX_COLOR_VALUE,
  getFullColorString,
  hsv2hex
} from '../../utilities/color/colors';
import * as stylesImport from './ColorPicker.scss';
const styles: any = stylesImport;

export interface IColorRectangleProps {
  componentRef?: () => void;
  color: IColor;
  minSize?: number;

  onSVChanged?(s: number, v: number): void;
}

export interface IColorPickerState {
  isAdjusting?: boolean;
  origin?: { x: number, y: number, color: IColor };
  color?: IColor;
  fullColorString?: string;
}

export class ColorRectangle extends BaseComponent<IColorRectangleProps, IColorPickerState> {
  public static defaultProps = {
    minSize: 220
  };

  private _root = createRef<HTMLDivElement>();

  constructor(props: IColorRectangleProps) {
    super(props);

    const { color } = this.props;

    this.state = {
      isAdjusting: false,
      origin: undefined,
      color: color,
      fullColorString: getFullColorString(color)
    };
  }

  public componentWillUnmount(): void {
    this._events.dispose();
  }

  public componentWillReceiveProps(newProps: IColorRectangleProps): void {
    const { color } = newProps;

    this.setState({
      color: color,
      fullColorString: getFullColorString(color)
    });
  }

  public render(): JSX.Element {
    const { minSize } = this.props;
    const { color, fullColorString } = this.state;

    return (
      <div ref={ this._root } className={ css('ms-ColorPicker-colorRect', styles.colorRect) } style={ { minWidth: minSize, minHeight: minSize, backgroundColor: fullColorString } } onMouseDown={ this._onMouseDown }>
        <div className={ css('ms-ColorPicker-light', styles.light) } />
        <div className={ css('ms-ColorPicker-dark', styles.dark) } />
        <div className={ css('ms-ColorPicker-thumb', styles.thumb) } style={ { left: color!.s + '%', top: (MAX_COLOR_VALUE - color!.v) + '%', backgroundColor: color!.str } } />
      </div>
    );
  }

  private _onMouseDown = (ev: React.MouseEvent<HTMLElement>): void => {
    this._events.on(window, 'mousemove', this._onMouseMove, true);
    this._events.on(window, 'mouseup', this._onMouseUp, true);

    this._onMouseMove(ev);
  }

  private _onMouseMove = (ev: React.MouseEvent<HTMLElement>): void => {
    const { color, onSVChanged } = this.props;

    if (!this._root.current) {
      return;
    }

    const rectSize = this._root.current.getBoundingClientRect();

    const sPercentage = (ev.clientX - rectSize.left) / rectSize.width;
    const vPercentage = (ev.clientY - rectSize.top) / rectSize.height;

    const newColor = assign({}, color, {
      s: Math.min(MAX_COLOR_SATURATION, Math.max(0, sPercentage * MAX_COLOR_SATURATION)),
      v: Math.min(MAX_COLOR_VALUE, Math.max(0, MAX_COLOR_VALUE - (vPercentage * MAX_COLOR_VALUE))),
    });

    newColor.hex = hsv2hex(newColor.h, newColor.s, newColor.v);
    newColor.str = newColor.a === 100 ? '#' + newColor.hex : `rgba(${newColor.r}, ${newColor.g}, ${newColor.b}, ${newColor.a / 100})`;

    this.setState({
      isAdjusting: true,
      color: newColor
    });

    if (onSVChanged) {
      onSVChanged(newColor.s, newColor.v);
    }
    ev.preventDefault();
    ev.stopPropagation();
  }

  private _onMouseUp = (ev: React.MouseEvent<HTMLElement>): void => {
    this._events.off();

    this.setState({
      isAdjusting: false,
      origin: undefined
    });
  }

}
