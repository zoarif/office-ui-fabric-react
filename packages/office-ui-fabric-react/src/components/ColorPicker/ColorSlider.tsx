import * as React from 'react';
import {
  BaseComponent,
  css,
  createRef
} from '../../Utilities';
import * as stylesImport from './ColorPicker.scss';
const styles: any = stylesImport;

export interface IColorSliderProps {
  componentRef?: () => void;
  minValue?: number;
  maxValue?: number;
  value?: number;
  thumbColor?: string;
  overlayStyle?: any;
  onChanged?: (newValue: number) => void;

  className?: string;
  style?: any;
}

export interface IColorSliderState {
  isAdjusting?: boolean;
  origin?: { x: number, originalValue: number };
  currentValue?: number;
}

export class ColorSlider extends BaseComponent<IColorSliderProps, IColorSliderState> {
  public static defaultProps = {
    minValue: 0,
    maxValue: 100,
    thumbColor: 'inherit',
    value: 0
  };

  private _root = createRef<HTMLDivElement>();

  constructor(props: IColorSliderProps) {
    super(props);

    const { value } = this.props;

    this.state = {
      isAdjusting: false,
      origin: undefined,
      currentValue: value
    };
  }

  public componentWillReceiveProps(newProps: IColorSliderProps): void {
    if (newProps && newProps.value) {
      this.setState({ currentValue: newProps.value });
    }
  }

  public render(): JSX.Element {
    const { className, minValue, maxValue, overlayStyle } = this.props;
    const { currentValue, isAdjusting } = this.state;

    const currentPercentage = 100 * (currentValue! - minValue!) / (maxValue! - minValue!);

    return (
      <div
        ref={ this._root }
        className={ css(
          'ms-ColorPicker-slider',
          styles.slider,
          className,
          isAdjusting && 'is-adjusting'
        ) }
        onMouseDown={ this._onMouseDown }
      >
        <div className={ css('ms-ColorPicker-sliderOverlay', styles.sliderOverlay) } style={ overlayStyle } />
        <div className={ css('ms-ColorPicker-thumb is-slider', styles.thumb, styles.thumbIsSlider) } style={ { left: currentPercentage + '%' } } />
      </div>
    );
  }

  private _onMouseDown = (ev: React.MouseEvent<HTMLElement>): void => {
    this._events.on(window, 'mousemove', this._onMouseMove, true);
    this._events.on(window, 'mouseup', this._onMouseUp, true);

    this._onMouseMove(ev);
  }

  private _onMouseMove = (ev: React.MouseEvent<HTMLElement>): void => {
    if (!this._root.current) {
      return;
    }

    const { onChanged, minValue, maxValue } = this.props;
    const rectSize = this._root.current.getBoundingClientRect();

    const currentPercentage = (ev.clientX - rectSize.left) / rectSize.width;
    const newValue = Math.min(maxValue!, Math.max(minValue!, currentPercentage * maxValue!));

    this.setState({
      isAdjusting: true,
      currentValue: newValue
    });

    if (onChanged) {
      onChanged(newValue);
    }

    ev.preventDefault();
    ev.stopPropagation();
  }

  private _onMouseUp = (ev: React.MouseEvent<HTMLElement>): void => {
    this._events.off();

    this.setState({
      isAdjusting: false,
      origin: undefined
    });
  }

}
