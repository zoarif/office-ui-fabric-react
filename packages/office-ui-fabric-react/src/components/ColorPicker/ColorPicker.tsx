import * as React from 'react';
import {
  BaseComponent,
  css
} from '../../Utilities';
import { IColorPickerProps } from './ColorPicker.types';
import { TextField } from '../../TextField';
import { ColorRectangle } from './ColorRectangle';
import { ColorSlider } from './ColorSlider';
import {
  MAX_COLOR_HUE,
  IColor,
  getColorFromString,
  getColorFromRGBA,
  updateA,
  updateH,
  updateSV
} from '../../utilities/color/colors';
import { FontClassNames } from '../../Styling';
import * as stylesImport from './ColorPicker.scss';
const styles: any = stylesImport;

export interface IColorPickerState {
  isOpen: boolean;
  color: IColor;
}

export class ColorPicker extends BaseComponent<IColorPickerProps, IColorPickerState> {
  public static defaultProps = {
    hexLabel: 'Hex',
    redLabel: 'Red',
    greenLabel: 'Green',
    blueLabel: 'Blue',
    alphaLabel: 'Alpha'
  };

  private hexText: TextField;
  private rText: TextField;
  private gText: TextField;
  private bText: TextField;
  private aText: TextField;

  constructor(props: IColorPickerProps) {
    super(props);

    this.state = {
      color: getColorFromString(props.color)
    } as IColorPickerState;
  }

  public componentWillReceiveProps(newProps: IColorPickerProps): void {
    if (newProps.color) {
      this._updateColor(getColorFromString(newProps.color));
    }
  }

  public render(): JSX.Element {
    const { color } = this.state;

    return (
      <div className={ css('ms-ColorPicker', styles.root) }>
        <div className={ css('ms-ColorPicker-panel', styles.panel) }>
          <ColorRectangle color={ color } onSVChanged={ this._onSVChanged } />
          <ColorSlider
            className={ css('is-hue', styles.colorSliderIsHue) }
            minValue={ 0 }
            maxValue={ MAX_COLOR_HUE }
            value={ color.h }
            onChanged={ this._onHChanged }
          />
          { !this.props.alphaSliderHidden && (
            <ColorSlider
              className={ css('is-alpha', styles.colorSliderIsAlpha) }
              overlayStyle={ { background: `linear-gradient(to right, transparent 0, ${color.str} 100%)` } }
              minValue={ 0 }
              maxValue={ 100 }
              value={ color.a }
              onChanged={ this._onAChanged }
            />) }
          <table className={ css('ms-ColorPicker-table', styles.table) } cellPadding='0' cellSpacing='0'>
            <thead>
              <tr className={ FontClassNames.small }>
                <td className={ styles.tableHexCell }>{ this.props.hexLabel }</td>
                <td>{ this.props.redLabel }</td>
                <td>{ this.props.greenLabel }</td>
                <td>{ this.props.blueLabel }</td>
                { !this.props.alphaSliderHidden && (
                  <td>{ this.props.alphaLabel }</td>) }
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  <TextField
                    className={ css('ms-ColorPicker-input', styles.input) }
                    value={ color.hex }
                    ref={ (ref) => this.hexText = ref! }
                    onBlur={ this._onHexChanged }
                    spellCheck={ false }
                  />
                </td>
                <td style={ { width: '18%' } }>
                  <TextField
                    className={ css('ms-ColorPicker-input', styles.input) }
                    onBlur={ this._onRGBAChanged }
                    value={ String(color.r) }
                    ref={ (ref) => this.rText = ref! }
                    spellCheck={ false }
                  />
                </td>
                <td style={ { width: '18%' } }>
                  <TextField
                    className={ css('ms-ColorPicker-input', styles.input) }
                    onBlur={ this._onRGBAChanged }
                    value={ String(color.g) }
                    ref={ (ref) => this.gText = ref! }
                    spellCheck={ false }
                  />
                </td>
                <td style={ { width: '18%' } }>
                  <TextField
                    className={ css('ms-ColorPicker-input', styles.input) }
                    onBlur={ this._onRGBAChanged }
                    value={ String(color.b) }
                    ref={ (ref) => this.bText = ref! }
                    spellCheck={ false }
                  />
                </td>
                { !this.props.alphaSliderHidden && (
                  <td style={ { width: '18%' } }>
                    <TextField
                      className={ css('ms-ColorPicker-input', styles.input) }
                      onBlur={ this._onRGBAChanged }
                      value={ String(color.a) }
                      ref={ (ref) => this.aText = ref! }
                      spellCheck={ false }
                    />
                  </td>
                ) }
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    );
  }

  private _onSVChanged = (s: number, v: number): void => {
    this._updateColor(updateSV(this.state.color, s, v));
  }

  private _onHChanged = (h: number): void => {
    this._updateColor(updateH(this.state.color, h));
  }

  private _onAChanged = (a: number): void => {
    this._updateColor(updateA(this.state.color, a));
  }

  private _onHexChanged = (): void => {
    this._updateColor(getColorFromString('#' + this.hexText.value));
  }

  private _onRGBAChanged = (): void => {
    this._updateColor(getColorFromRGBA({
      r: Number(this.rText.value),
      g: Number(this.gText.value),
      b: Number(this.bText.value),
      a: Number((this.aText && this.aText.value) || 100)
    }));
  }

  private _updateColor(newColor?: IColor): void {
    if (!newColor) {
      return;
    }

    const { onColorChanged } = this.props;

    if (newColor.str !== this.state.color.str) {
      this.setState({
        color: newColor
      } as IColorPickerState, () => {
        if (onColorChanged) {
          onColorChanged(newColor.str);
        }
      });
    }
  }
}
