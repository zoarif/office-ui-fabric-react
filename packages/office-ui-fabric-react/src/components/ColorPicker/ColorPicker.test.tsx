/* tslint:disable:no-unused-variable */
import * as React from 'react';
import * as ReactDOM from 'react-dom';
/* tslint:enable:no-unused-variable */
import * as ReactTestUtils from 'react-dom/test-utils';
import * as renderer from 'react-test-renderer';

import { ColorPicker } from './ColorPicker';

describe('ColorPicker', () => {
  it('renders ColorPicker correctly', () => {
    const component = renderer.create(
      <ColorPicker color='#FFFFFF' />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Props are correctly parsed', () => {
    const component = ReactTestUtils.renderIntoDocument(
      <ColorPicker color='#FFFFFF' />
    ) as ColorPicker;

    expect(component.state.color.hex).toEqual('ffffff');
  });

  it('Reacts to props changes', () => {
    const component = ReactTestUtils.renderIntoDocument(
      <ColorPicker color='#FFFFFF' />
    ) as ColorPicker;

    component.componentWillReceiveProps({ color: '#AEAEAE' });
    expect(component.state.color.hex).toEqual('aeaeae');
  });

  it('onColorChange is called', () => {
    let color = '#FFFFFF';
    const onColorChanged = (str: string): void => {
      color = str;
    };

    const component = ReactTestUtils.renderIntoDocument(
      <ColorPicker
        color={ color }
        onColorChanged={ onColorChanged }
      />
    ) as ColorPicker;

    const newColor = '#AEAEAE';
    component.componentWillReceiveProps({ color: newColor });

    expect(component.state.color.hex).toEqual('aeaeae');
    expect(color).toEqual(newColor);
  });

  it('Hides alpha control slider', () => {
    const component = ReactTestUtils.renderIntoDocument(
      <ColorPicker color='#FFFFFF' alphaSliderHidden={ true } />
    ) as ColorPicker;

    const renderedDOM = ReactDOM.findDOMNode(component as React.ReactInstance) as Element;
    const alphaSlider = renderedDOM.querySelector('.is-alpha');
    const alphaTableHeader = renderedDOM.querySelector('.ms-ColorPicker-table > thead > tr > td:nth-child(5)');
    const alphaTableInput = renderedDOM.querySelector('.ms-ColorPicker-table > tbody> tr > td:nth-child(5)');

    expect(alphaSlider).toBeNull();
    expect(alphaTableHeader).toBeNull();
    expect(alphaTableInput).toBeNull();
  });

  it('Renders default RGBA/Hex strings', () => {
    const component = ReactTestUtils.renderIntoDocument(
      <ColorPicker color='#FFFFFF' />
    ) as ColorPicker;

    const renderedDOM = ReactDOM.findDOMNode(component as React.ReactInstance) as Element;
    const tableHeaders = renderedDOM.querySelectorAll('.ms-ColorPicker-table > thead > tr > td') as NodeListOf<HTMLTableDataCellElement>;

    const hexTableHeader = tableHeaders[0];
    expect(hexTableHeader.textContent).toEqual(ColorPicker.defaultProps.hexLabel);

    const redTableHeader = tableHeaders[1];
    expect(redTableHeader.textContent).toEqual(ColorPicker.defaultProps.redLabel);

    const greenTableHeader = tableHeaders[2];
    expect(greenTableHeader.textContent).toEqual(ColorPicker.defaultProps.greenLabel);

    const blueTableHeader = tableHeaders[3];
    expect(blueTableHeader.textContent).toEqual(ColorPicker.defaultProps.blueLabel);

    const alphaTableHeader = tableHeaders[4];
    expect(alphaTableHeader.textContent).toEqual(ColorPicker.defaultProps.alphaLabel);
  });

  it('Renders custom RGBA/Hex strings', () => {
    const customHexLabel = 'Custom Hex';
    const customRedLabel = 'Custom Red';
    const customGreenLabel = 'Custom Green';
    const customBlueLabel = 'Custom Blue';
    const customAlphaLabel = 'Custom Alpha';

    const component = ReactTestUtils.renderIntoDocument(
      <ColorPicker
        color='#FFFFFF'
        hexLabel={ customHexLabel }
        redLabel={ customRedLabel }
        greenLabel={ customGreenLabel }
        blueLabel={ customBlueLabel }
        alphaLabel={ customAlphaLabel }
      />
    ) as ColorPicker;

    const renderedDOM = ReactDOM.findDOMNode(component as React.ReactInstance) as Element;
    const tableHeaders = renderedDOM.querySelectorAll('.ms-ColorPicker-table > thead > tr > td') as NodeListOf<HTMLTableDataCellElement>;

    const hexTableHeader = tableHeaders[0];
    expect(hexTableHeader.textContent).toEqual(customHexLabel);

    const redTableHeader = tableHeaders[1];
    expect(redTableHeader.textContent).toEqual(customRedLabel);

    const greenTableHeader = tableHeaders[2];
    expect(greenTableHeader.textContent).toEqual(customGreenLabel);

    const blueTableHeader = tableHeaders[3];
    expect(blueTableHeader.textContent).toEqual(customBlueLabel);

    const alphaTableHeader = tableHeaders[4];
    expect(alphaTableHeader.textContent).toEqual(customAlphaLabel);
  });
});
