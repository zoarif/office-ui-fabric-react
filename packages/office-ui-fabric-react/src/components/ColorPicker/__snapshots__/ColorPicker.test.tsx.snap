// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker renders ColorPicker correctly 1`] = `
<div
  className="ms-ColorPicker"
>
  <div
    className="ms-ColorPicker-panel"
  >
    <div
      className="ms-ColorPicker-colorRect"
      onMouseDown={[Function]}
      style={
        Object {
          "backgroundColor": "#ff0000",
          "minHeight": 220,
          "minWidth": 220,
        }
      }
    >
      <div
        className="ms-ColorPicker-light"
      />
      <div
        className="ms-ColorPicker-dark"
      />
      <div
        className="ms-ColorPicker-thumb"
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "left": "0%",
            "top": "0%",
          }
        }
      />
    </div>
    <div
      className="ms-ColorPicker-slider is-hue"
      onMouseDown={[Function]}
    >
      <div
        className="ms-ColorPicker-sliderOverlay"
        style={undefined}
      />
      <div
        className="ms-ColorPicker-thumb is-slider"
        style={
          Object {
            "left": "0%",
          }
        }
      />
    </div>
    <div
      className="ms-ColorPicker-slider is-alpha"
      onMouseDown={[Function]}
    >
      <div
        className="ms-ColorPicker-sliderOverlay"
        style={
          Object {
            "background": "linear-gradient(to right, transparent 0, #FFFFFF 100%)",
          }
        }
      />
      <div
        className="ms-ColorPicker-thumb is-slider"
        style={
          Object {
            "left": "100%",
          }
        }
      />
    </div>
    <table
      cellPadding="0"
      cellSpacing="0"
      className="ms-ColorPicker-table"
    >
      <thead>
        <tr
          className=

              {
                -moz-osx-font-smoothing: grayscale;
                -webkit-font-smoothing: antialiased;
                font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                font-size: 12px;
                font-weight: 400;
              }
        >
          <td
            className={undefined}
          >
            Hex
          </td>
          <td>
            Red
          </td>
          <td>
            Green
          </td>
          <td>
            Blue
          </td>
          <td>
            Alpha
          </td>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <div
              className="ms-TextField ms-ColorPicker-input"
            >
              <div
                className="ms-TextField-wrapper"
              >
                <div
                  className="ms-TextField-fieldGroup"
                >
                  <input
                    aria-describedby={undefined}
                    aria-invalid={false}
                    aria-label={undefined}
                    className="ms-TextField-field"
                    id="TextField0"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    spellCheck={false}
                    type="text"
                    value="ffffff"
                  />
                </div>
              </div>
            </div>
          </td>
          <td
            style={
              Object {
                "width": "18%",
              }
            }
          >
            <div
              className="ms-TextField ms-ColorPicker-input"
            >
              <div
                className="ms-TextField-wrapper"
              >
                <div
                  className="ms-TextField-fieldGroup"
                >
                  <input
                    aria-describedby={undefined}
                    aria-invalid={false}
                    aria-label={undefined}
                    className="ms-TextField-field"
                    id="TextField2"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    spellCheck={false}
                    type="text"
                    value="255"
                  />
                </div>
              </div>
            </div>
          </td>
          <td
            style={
              Object {
                "width": "18%",
              }
            }
          >
            <div
              className="ms-TextField ms-ColorPicker-input"
            >
              <div
                className="ms-TextField-wrapper"
              >
                <div
                  className="ms-TextField-fieldGroup"
                >
                  <input
                    aria-describedby={undefined}
                    aria-invalid={false}
                    aria-label={undefined}
                    className="ms-TextField-field"
                    id="TextField4"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    spellCheck={false}
                    type="text"
                    value="255"
                  />
                </div>
              </div>
            </div>
          </td>
          <td
            style={
              Object {
                "width": "18%",
              }
            }
          >
            <div
              className="ms-TextField ms-ColorPicker-input"
            >
              <div
                className="ms-TextField-wrapper"
              >
                <div
                  className="ms-TextField-fieldGroup"
                >
                  <input
                    aria-describedby={undefined}
                    aria-invalid={false}
                    aria-label={undefined}
                    className="ms-TextField-field"
                    id="TextField6"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    spellCheck={false}
                    type="text"
                    value="255"
                  />
                </div>
              </div>
            </div>
          </td>
          <td
            style={
              Object {
                "width": "18%",
              }
            }
          >
            <div
              className="ms-TextField ms-ColorPicker-input"
            >
              <div
                className="ms-TextField-wrapper"
              >
                <div
                  className="ms-TextField-fieldGroup"
                >
                  <input
                    aria-describedby={undefined}
                    aria-invalid={false}
                    aria-label={undefined}
                    className="ms-TextField-field"
                    id="TextField8"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    spellCheck={false}
                    type="text"
                    value="100"
                  />
                </div>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
